src/kOS/Suffixed/Direction.cs

```diff
@@ -174,6 +174,38 @@ private void DirectionInitializeSuffixes()
         {
             return new Direction(a.rotation.Inverse());
         }
+
+        public override bool Equals(object obj)
+        {
+            Direction d = obj as Direction;
+            if (d != null)
+            {
+                // Check the equality of the quaternion components
+                if (d.rotation.x == rotation.x &&
+                    d.rotation.y == rotation.y &&
+                    d.rotation.z == rotation.z)
```

Comment:
`UnityEngine.Quaternion` already has its own `Equals()` operator method defined.  It might be a good idea to just defer to using that instead of re-implementing it (and I had a peek in ILSpy and it turns out what you're doing here i almost the same thing Unity is doing for Equals, with the exception that you only compare the `x`, `y`, and `z` and Unity is comparing all 4 components (including the `w`).)